pandas/io/gcs.py

```diff
@@ -0,0 +1,26 @@
+""" GCS support for remote file interactivity """
+from pandas import compat
+try:
+    from google.auth.exceptions import GoogleAuthError
+    import gcsfs
+except ImportError:
+    raise ImportError("The gcsfs library is required to handle GCS files")
+
+
+def get_filepath_or_buffer(filepath_or_buffer, encoding=None,
+                           compression=None, mode=None):
+
+    if mode is None:
+        mode = 'rb'
+
+    gcsfs_logger_disabled = gcsfs.core.logger.disabled
+    gcsfs.core.logger.disabled = True
+    fs = gcsfs.GCSFileSystem()
+    try:
+        filepath_or_buffer = fs.open(filepath_or_buffer, mode)
+    except (compat.FileNotFoundError, GoogleAuthError, gcsfs.utils.HtmlError):
+        fs = gcsfs.GCSFileSystem(token='anon')
+        filepath_or_buffer = fs.open(filepath_or_buffer, mode)
+    finally:
+        gcsfs.core.logger.disabled = gcsfs_logger_disabled
```

Comment:
If the anon version fails, this line is not hit, correct? It would have to be wrapped in a `try / finally` as well (or ideally the logger stuff would be removed).